packages/store/ts/config/v2/README.md

```diff
@@ -0,0 +1,47 @@
+# Config conventions
+
+These are the types and functions that should be placed in a file called `x.ts`
+
+```ts
+/**
+ * validateX returns input if the input is valid and expected type otherwise.
```

Comment:
do we wanna add anything about _why_ this is the approach we use? returning valid input makes sense, returning expected type is less obvious to me

Reply:
returning the expected type so that typescript compares the input to the expected type!

if the function is `defineConfig<input>(input: validateInput<input>)` then what you're passing in has to match `validateInput<input>`. If `validateInput<input>` is happy it returns `input`, so it matches; if it's not happy it returns what it expected, so the input param basically becomes `defineConfig<input>(input: Expected)`, typescript will tell you why `input` doesn't match `Expected`

Reply:
great, do we wanna document that here?